src/api/c/err_common.hpp

```diff
@@ -207,3 +207,7 @@ void print_error(const std::string &msg);
 
 static const int MAX_ERR_SIZE = 1024;
 std::string& get_global_error_string();
+
+#ifdef AF_OPENCL
+#include <err_opencl.hpp>
+#endif
```

Comment:
This is a circular dependency. `err_opencl.hpp` is including `err_common.hpp`. Not sure if this is a good idea. I think the CATCHALL should go in a separate file.